staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -473,18 +473,25 @@ func (config *inClusterClientConfig) ClientConfig() (*restclient.Config, error)
 		return nil, err
 	}
 
-	// in-cluster configs only takes a host, token, or CA file
-	// if any of them were individually provided, ovewrite anything else
+	// in-cluster configs only takes --as, --as-group, --request-timeout
```

Comment:
This is a good question.
Because overrides this value here doesn't make any sense.
When we use -s to override config.overrides.ClusterInfo.Server, we will use mergedConfig rather than inClusterConfig
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/clientcmd/merged_client_builder.go#L116
So override this value here doesn't make any sense, this logic will never run.